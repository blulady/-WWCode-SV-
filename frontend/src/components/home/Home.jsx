import React from "react";
import { useNavigate } from "react-router-dom";
import { useAuthContext } from "../../context/auth/AuthContext";
import ContainerWithNav from "../layout/ContainerWithNav";
import styles from "./Home.module.css";
import { NAVITEMS } from "../../navitems";

/*
 * Home page - contains header and Chapter Members link in the body
 */

const Home = () => {
  const navigate = useNavigate();
  const handleClick = (pageId) => {
    navigate(`/${pageId}/members`);
  };
  const { userInfo } = useAuthContext();

  return (
    <>
      {userInfo && (
        <ContainerWithNav>
          <div className={`${styles["home-container"]} container`}>
            <div className="row">
              {NAVITEMS.map((item) => {
                return (
                  <>
                    {item.pageId === "directors" && userInfo.highest_role !== 'DIRECTOR' ? <></> :
                      <div key={item.pageId} className={`${styles["home-card"]} ${styles[item.name.toLowerCase().replaceAll(" ", "-")]} col-12 col-md-4`} onClick={() => handleClick(item.pageId)}>
                        <div className={styles.cardimgtop}></div>
                        <div className={`${styles.cardbody} d-flex justify-content-center align-items-center`}>
                          {item.name}
                        </div>
                      </div>
                    }
                  </>
                )
              })}
            </div>
          </div>
        </ContainerWithNav>
      )}
    </>
  );
};

export default Home;
