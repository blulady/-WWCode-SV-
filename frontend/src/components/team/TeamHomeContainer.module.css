.team-home-container {
  position: relative;
  height: 1132px;
  max-height: 1132px;
  overflow-y: hidden;
  width: 1078px;
  margin: 0 auto;
}

.team-home-container-inner {
  height: 1074px;
}

.team-home-tab-container {
  height: 100px;
}

.tab-button {
  min-width: 165px;
  height: 51px;
  padding: 0 15px;
}

.tab-button > a {
  color: black;
}

.tab-button-selected {
  background: rgba(130, 130, 130, 0.1);
  border-radius :8px;
  outline: none !important;
}

.team-home-tab-contents {
  position: relative;
  height: 1132px;
  max-height: 1132px;
  overflow-y: hidden;
  width: 1078px;
  margin: 0 auto;
  padding-top: 25px;
}

@media only screen and (max-width: 576px) {
  .team-home-container {
    position: relative;
    /*height: calc(100vh - 60px);
    max-height: calc(100vh - 60px);
    overflow-y: scroll; */
    width: 95vw;
    margin: 0 auto;
    height: auto;
  }

  .team-home-tab-contents {
    width: auto;
    height: 100%;
    overflow-y: scroll;
    padding-top: 15px;
  }
}