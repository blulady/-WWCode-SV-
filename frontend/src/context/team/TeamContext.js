import { createContext, useContext } from 'react';

const TeamContext = createContext();

export const useTeamContext = () => useContext(TeamContext);
export default TeamContext;