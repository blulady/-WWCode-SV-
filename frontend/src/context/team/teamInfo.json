[
  {
    "pages": [
      {
        "pageId": "members",
        "label": "Chapter Members"
      },
      {
        "pageId": "resources",
        "label": "Volunteer Resources"
      }
    ],
    "slug": "volunteer_resource",
    "enabled": true
  },
  {
    "pages": [
      {
        "pageId": "members",
        "label": "Event Volunteers"
      },
      {
        "pageId": "resources",
        "label": ""
      }
    ],
    "slug": "",
    "enabled": false
  },
  {
    "pages": [
      {
        "pageId": "members",
        "label": "Hackathon Volunteers"
      },
      {
        "pageId": "resources",
        "label": ""
      }
    ],
    "slug": "",
    "enabled": false
  },
  {
    "pages": [
      {
        "pageId": "members",
        "label": "Host Management"
      },
      {
        "pageId": "resources",
        "label": ""
      }
    ],
    "slug": "",
    "enabled": false
  },
  {
    "pages": [
      {
        "pageId": "members",
        "label": "Partnership Management"
      },
      {
        "pageId": "resources",
        "label": ""
      }
    ],
    "slug": "",
    "enabled": false
  },
  {
    "pages": [
      {
        "pageId": "members",
        "label": "Social Media Team"
      },
      {
        "pageId": "resources",
        "label": "Sodial Media Resources"
      }
    ],
    "slug": "social_media_resource",
    "enabled": true
  },
  {
    "pages": [
      {
        "pageId": "members",
        "label": "Tech Event Volunteers"
      },
      {
        "pageId": "resources",
        "label": ""
      }
    ],
    "slug": "",
    "enabled": false
  },
  {
    "pages": [
      {
        "pageId": "members",
        "label": "Volunteer Management"
      },
      {
        "pageId": "resources",
        "label": ""
      }
    ],
    "slug": "",
    "enabled": false
  },
  {
    "pages": [
      {
        "pageId": "members",
        "label": "Tech Bloggers"
      },
      {
        "pageId": "resources",
        "label": "Tech Bloggers Resources"
      }
    ],
    "slug": "tech_bloggers",
    "enabled": true
  }
]
