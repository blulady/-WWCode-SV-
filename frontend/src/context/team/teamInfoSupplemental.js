export const directorsInfo = {id: 9, 
  name: 'Directors',
  pages:  [
    {
      "pageId": "members",
      "label": "Directors",
      "isDirectorOnly": true
    },
    {
      "pageId": "notes",
      "label": "Director Meeting Notes",
      "isDirectorOnly": true
    },
    {
      "pageId": "business",
      "label": "Director Business",
      "isDirectorOnly": true
    },
    {
      "pageId": "event-calendar",
      "label": "Event Calendar",
      "isDirectorOnly": true
    }
  ], 
  slug: "directors_resource",
  enabled: true}
