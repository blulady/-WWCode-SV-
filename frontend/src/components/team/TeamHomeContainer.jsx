import React, { useState, useContext } from 'react'
import ContainerWithNav from "../layout/ContainerWithNav";
import styles from "./TeamHomeContainer.module.css";
import cx from "classnames";
import { NavLink, Outlet, useParams, useLocation } from 'react-router-dom';
import { useTeamContext } from '../../context/team/TeamContext';
import AuthContext from "../../context/auth/AuthContext";

const TeamHomeContainer = (props) => {
  const params = useParams();
  const team = parseInt(params.team);
  const { teams } = useTeamContext();
  const teamInfo = teams[team];
  const location = useLocation();
  let pageId = 0;
  teamInfo.pages.forEach((p, index) => {
    if (location.pathname.indexOf(p.pageId) > -1) {
      pageId = index;
    }
  });
  const [currentPage, setCurrentPage] = useState(pageId);



  const { userInfo } = useContext(AuthContext);
  const isDirector = userInfo.role === "DIRECTOR";

  return (
    <ContainerWithNav>
      <div
        id='teamHomeContainer'
        className={cx(styles["team-home-container"], "d-flex flex-column")}
      >
        <div className={styles["team-home-container-inner"]}>
          <div className={styles["team-home-tab-container"] + " d-flex align-items-end"}>
            {
              teamInfo.pages.map((page, i) => {
                if (!page.isDirectorOnly || isDirector) {
                  return (
                  
                    <NavLink to={"/team/" + team + "/" + page.pageId} key={i}>
                      <button className={styles["tab-button"] + ((i === currentPage) ? " " + styles["tab-button-selected"] : '')} onClick={() => setCurrentPage(i)}>
                        {page.label}</button></NavLink>
                  )
                }
              }
              )
            }
          </div>
          <div
            className={cx(styles["team-home-tab-contents"], "d-flex flex-column")}
          >
            <Outlet />
          </div>
        </div>
      </div>
    </ContainerWithNav>
  )
};
export default TeamHomeContainer;