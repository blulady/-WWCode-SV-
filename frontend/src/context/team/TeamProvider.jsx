import React, { useEffect, useState } from "react";
import TeamContext from "./TeamContext";
import WwcApi from "../../WwcApi"

// TODO: Temp variable to filter out teams not yet developed
const activeTeams = [5]

const TeamProvider = ({ children }) => {
    const [teams, setTeams] = useState([]);
    const [fetching, setFetching] = useState(true); // TODO: Is there a better way to delay rendering??

    useEffect(() => {
        const fetchTeams = async () => {
            let _teams = [];
            try {
                _teams = await WwcApi.getTeams();
            } catch (e) {
                console.log(e);
            }
            const teamCount = _teams.length - 1
            _teams = _teams.filter((t) => activeTeams.indexOf(t.id) > -1 );
            // TODO: Remove me once backend adds Tech Bloggers team
            _teams.push({ id: teamCount, name: "Tech Bloggers" })
            setTeams([...teams, ..._teams]);
            setFetching(false);
        }
        fetchTeams();
    }, []);

    return (
        <TeamContext.Provider value={ { teams: (fetching ? [] : teams) } }>{children}</TeamContext.Provider>  
    );
}

export default TeamProvider;