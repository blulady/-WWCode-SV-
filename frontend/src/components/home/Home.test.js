import React from 'react';
import { render } from '@testing-library/react';
import Home from './Home';
import * as TeamContext from '../../context/team/TeamContext';

jest.mock('react-router-dom', () => ({
    useNavigate: () => ({
      navigate: jest.fn()
    })
}));

jest.mock('react', () => {
    const ActualReact = jest.requireActual('react');
    return {
        ...ActualReact,
        useContext: () => ({
            userInfo: {
                email: 'director@example.com',
                first_name: 'John',
                id: 1,
                last_name: 'Smith',
                role: 'DIRECTOR'
            },
            handleRemoveAuth: jest.fn()
        })
    };
});

describe('Home', () => {
    test('it renders without crashing', () => {
        const contextTeams = { teams: [{ id: 1, name: 'Team1' }] };
        jest.spyOn(TeamContext, 'useTeamContext')
        .mockImplementation(() => contextTeams);
        const { container } = render(<Home />);
        expect(container).toMatchSnapshot();
    });
});