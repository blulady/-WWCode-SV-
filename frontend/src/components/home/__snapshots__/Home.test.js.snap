// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home it renders without crashing 1`] = `
<div>
  <div
    class="container-fluid container-fluid"
  >
    <nav
      class="container-navbar"
    >
      <div
        class="container-inner"
      >
        <div
          class="navbar-brand navbar-wwc-logo"
        >
          <div
            class="logo-png"
          />
        </div>
        <div
          class="navbar-items "
        >
          <ul
            class="navbar-items-ul"
          >
            <li
              class="navbar-text navbar-items-wwc-title"
            >
              Chapter Tools
            </li>
            <li
              class="navbar-items-user-button"
            >
              <div
                aria-expanded="false"
                aria-haspopup="true"
                class="user-button"
                data-toggle="dropdown"
                id="navbarDropdown"
                role="button"
              >
                <span>
                  J
                </span>
              </div>
              <div
                aria-labelledby="navbarDropdown"
                class="dropdown-menu dropdown-menu"
              >
                <p
                  class="dropdown-item signed-user dropdown-item"
                >
                  Signed-in as
                </p>
                <p
                  class="font-weight-bold dropdown-item signed-user dropdown-item"
                >
                  director@example.com
                </p>
                <button
                  class="dropdown-item dropdown-item"
                >
                  Your Profile
                </button>
                <hr
                  class="dropdown-divider dropdown-hr"
                />
                <button
                  class="dropdown-item item-logout dropdown-item"
                >
                  Log Out
                </button>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div
      class="container-body"
    >
      <div
        class="home-container container"
      >
        <div
          class="row"
        >
          <div
            class="home-card chapter-members col-12 col-md-4"
          >
            <div
              class="cardimgtop"
            />
            <div
              class="cardbody d-flex justify-content-center align-items-center"
            >
              Chapter Members
            </div>
          </div>
          <div
            class="home-card team1 col-12 col-md-4"
          >
            <div
              class="cardimgtop"
            />
            <div
              class="cardbody d-flex justify-content-center align-items-center"
            >
              Team1
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
